core/templates/dev/head/pages/exploration_editor/editor_tab/UnresolvedAnswersOverviewDirective.js

```diff
@@ -25,72 +25,18 @@ oppia.directive('unresolvedAnswersOverview', [
         '/pages/exploration_editor/editor_tab/' +
         'unresolved_answers_overview_directive.html'),
       controller: [
-        '$scope', 'EditorStateService', 'ExplorationStatesService',
-        'StateRulesStatsService', 'ExplorationRightsService',
-        function(
-            $scope, EditorStateService, ExplorationStatesService,
-            StateRulesStatsService, ExplorationRightsService) {
-          var MAXIMUM_UNRESOLVED_ANSWERS = 5;
-          var MINIMUM_UNRESOLVED_ANSWER_FREQUENCY = 2;
-
-          $scope.unresolvedAnswersData = null;
-          $scope.latestRefreshDate = new Date();
-          $scope.unresolvedAnswersOverviewIsShown = false;
-
-          $scope.computeUnresolvedAnswers = function() {
-            var state = ExplorationStatesService.getState(
+        '$scope', 'EditorStateService', 'StateTopAnswersStatsService',
+        function($scope, EditorStateService, StateTopAnswersStatsService) {
+          $scope.getStateStats = function() {
+            return StateTopAnswersStatsService.getStateStats(
               EditorStateService.getActiveStateName()
-            );
-
-            if (!StateRulesStatsService.stateSupportsIssuesOverview(state)) {
-              $scope.unresolvedAnswersData = [];
-            } else {
-              StateRulesStatsService.computeStateRulesStats(
-                state
-              ).then(function(stats) {
-                var calculatedUnresolvedAnswersData = [];
-
-                for (var i = 0; i < stats.visualizations_info.length; ++i) {
-                  var vizInfo = stats.visualizations_info[i];
-                  if (!vizInfo.addressed_info_is_supported) {
-                    continue;
-                  }
-
-                  // NOTE: vizInfo.data is already sorted in descending order by
-                  // frequency.
-                  for (var j = 0; j < vizInfo.data.length; ++j) {
-                    var answer = vizInfo.data[j];
-                    if (answer.is_addressed ||
-                        answer.frequency <
-                          MINIMUM_UNRESOLVED_ANSWER_FREQUENCY) {
-                      continue;
-                    }
-
-                    calculatedUnresolvedAnswersData.push(answer);
-                    if (calculatedUnresolvedAnswersData.length >=
-                          MAXIMUM_UNRESOLVED_ANSWERS) {
-                      break;
-                    }
-                  }
-
-                  // Will only take the answers from first eligible
-                  // visualization.
-                  break;
-                }
-
-                $scope.unresolvedAnswersData = calculatedUnresolvedAnswersData;
-                $scope.latestRefreshDate = new Date();
-              });
-            }
+            ).filter(function(answerStats) {
+              return !answerStats.isAddressed;
+            });
+          };
+          $scope.unresolvedAnswersOverviewIsShown = function() {
```

Comment:
Why make this a function?

Reply:
Because the stats may not have been initialized by the time this directive loads (since we async it after the exploration is ready).

Reply:
OK, then let's rename to $scope.isUnresolvedAnswersOverviewShown() (start with verb)

Reply:
Done.